modules/tinymce/src/themes/silver/main/ts/ui/core/complex/AlignSelect.ts

```diff
@@ -47,13 +46,17 @@ const getSpec = (editor: Editor): SelectSpec => {
 
   const dataset = buildBasicStaticDataset(alignMenuItems);
 
+  const onAction = (rawItem: FormatterFormatItem) => () =>
+    Arr.find(alignMenuItems, (item) => item.format === rawItem.format)
+      .each((item) => editor.execCommand(item.command));
```

Comment:
Why did you decide to change this from onActionToggleFormat? Among other things, afaik execCommand doesn't call undoManager.transact(), so aren't we losing the undo stack entry? Also are you able to apply/unapply the style properly if you e.g. apply the style to some content, edit and style other content, then come back to the original content and try to remove the style?

Reply:
This is the code used by the toolbar buttons (see `AlignmentButtons.ts`), the two were different and the previous version doesn't remove the formats causing both center and right align for example to be applied at once.

You're probably right about the missing undo manager transaction, but that also means all the toolbar buttons currently aren't doing that either.

Reply:
can we at least add an undo transaction here (or even better, in the command handling code) so there isn't a regression, even if other buttons have already regressed?

Reply:
Agreed, I didn't mean it to come across as though we should leave it. It definitely should be wrapped in an undo transaction.

Reply:
I looked into this more after discussing more with @ltrouton and it turns out the UndoManager listens for both the `BeforeExecCommand` and `ExecCommand` events and adds levels accordingly, so core was already handling this. See https://github.com/tinymce/tinymce/blob/develop/modules/tinymce/src/core/main/ts/undo/Setup.ts#L28